chore(release): 🚀 publish v33.0.0 (#1240)
traefik/traefik-helm-chart: traefik/Changelog.md

**Release date:** 2024-10-30

* fix: 🐛 http3 with internal service
* fix: use correct children indentation for logs.access.filters
* fix(schema): :bug: targetPort can also be a string
* fix(certificateResolvers)!: :boom: :bug: use same syntax in Chart and in Traefik
* fix(Traefik)!: :boom: set 8080 as default port for `traefik` entrypoint
* fix(Traefik Hub): RBAC for distributedAcme
* fix(Kubernetes Ingress)!: :boom: :sparkles: enable publishedService by default
* fix(Gateway API): :bug: add missing required RBAC for v3.2 with experimental Channel
* fix(Env Variables)!: allow extending env without overwrite
* feat(deps): update traefik docker tag to v3.2.0
* feat(deps): update traefik docker tag to v3.1.6
* feat(Traefik): ✨ support Gateway API statusAddress
* feat(Traefik Proxy): CRDs for v3.2+
* feat(Gateway API): :sparkles: standard install CRD v1.2.0
* feat(Gateway API): :sparkles: add infrastructure in the values
* chore: allow TRACE log level
* chore(release): 🚀 publish v33.0.0
* Update topology spread constraints comments

**Upgrade Notes**

There are multiple breaking changes in this release:

> 1. Default port of `traefik` entrypoint is changed from 9000 to 8080, like Traefik Proxy default

Reviewer: ```suggestion
1. The default port of `traefik` entrypoint has changed from `9000` to `8080`, just like the Traefik Proxy default port
```

  * You _may_ have to update probes accordingly (or set this port back to 9000)

> 2. publishedService is enabled by default on Ingress provider

Reviewer: ```suggestion
2. `publishedService` is enabled by default on Ingress provider
```

  * You _can_ disable it, if needed

> 3. `POD_NAME` and `POD_NAMESPACE` environnement variables are now set by default, without values.

Reviewer: ```suggestion
3. The `POD_NAME` and `POD_NAMESPACE` environment variables are now set by default, without values.
```

  * It is no longer necessary to add them in values and so, it can be removed from user values.
4. In _values_, **certResolvers** specific syntax has been reworked to align with Traefik Proxy syntax.
  * PR [#1214](https://github.com/traefik/traefik-helm-chart/pull/1214) contains a complete before / after example on how to update _values_
5. Traefik Proxy 3.2 supports Gateway API v1.2

>   * The crds of this version comes with Gateway API CRD v1.2 of standard channel.

Reviewer: ```suggestion
  * The CRDs of this version comes with Gateway API CRD v1.2 of standard channel.
```

>   * CRDs needs to be updated, as documented in the README.

Reviewer: ```suggestion
  * The CRDs needs to be updated, as documented in the README.
```

  * It is recommended to check that other software using Gateway API on your cluster are compatible

> :information_source: A separate helm chart, just for CRDs, is actually considered for future release. See PR [#1123](https://github.com/traefik/traefik-helm-chart/pull/1223)

Reviewer: ```suggestion
:information_source: A separate helm chart, just for CRDs, is being considered for a future release. See PR [#1123](https://github.com/traefik/traefik-helm-chart/pull/1223)
```

     exposedPort: 443
     ## -- Different target traefik port on the cluster, useful for IP type LB
-    targetPort:  # @schema type:[integer, null]; minimum:0
+    targetPort:  # @schema type:[string, integer, null]; minimum:0
     ## -- The port protocol (TCP/UDP)
     protocol: TCP
     # -- See [upstream documentation](https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport)
@@ -780,8 +787,8 @@ autoscaling:
 
 persistence:
   # -- Enable persistence using Persistent Volume Claims
-  # ref: http://kubernetes.io/docs/user-guide/persistent-volumes/
-  # It can be used to store TLS certificates, see `storage` in certResolvers
+  # ref: http://kubernetes.io/docs/user-guide/persistent-volumes/.
+  # It can be used to store TLS certificates along with `certificatesResolvers.<name>.acme.storage`  option
   enabled: false
   name: data
   existingClaim: ""
@@ -797,7 +804,7 @@ persistence:
 # -- Certificates resolvers configuration.
 # Ref: https://doc.traefik.io/traefik/https/acme/#certificate-resolvers
 # See EXAMPLES.md for more details.
-certResolvers: {}
+certificatesResolvers: {}
 
 # -- If hostNetwork is true, runs traefik in the host network namespace

>  # To prevent unschedulabel pods due to port collisions, if hostNetwork=true

Reviewer: ```suggestion
 # To prevent unschedulable pods due to port collisions, if hostNetwork=true
```

Author: Good catch ! This one has to be done on values.yaml, not this PR. It will be for the next release.

Reviewer: alright!

Author: I opened this PR  to fix it: https://github.com/traefik/traefik-helm-chart/pull/1241
